sklearn/metrics/_classification.py

```diff
@@ -3887,16 +3944,17 @@ def d2_brier_score(
         transformed_labels, y_proba = _validate_multiclass_probabilistic_prediction(
             y_true, y_proba, sample_weight, labels
         )
-    y_proba_null = np.average(transformed_labels, axis=0, weights=sample_weight)
-    y_proba_null = np.tile(y_proba_null, (len(y_true), 1))
+    transformed_labels = xp.astype(transformed_labels, y_proba.dtype, copy=False)
```

Comment:
Maybe we could add a `dtype` param to the one hot encoding helpers.

Reply:
But that would then create confusions when we don't need such a change. I think it might make sense to do this where it's required.